### ISSUE TITLE
stat module returns mode 420 for a file that is 0644 

### ISSUE THREAD

**Issue by jctanner** [ansible/ansible#6080]
##### Issue Type:

Bug Report
##### Ansible Version:

```
ansible 1.5 (devel ee6a0359f5) last updated 2014/02/19 16:45:25 (GMT -400)
```
##### Environment:

ubuntu 12.04 on both hosts
##### Summary:

A file with 0644 permissions is reported as 420 by the stat module.
##### Steps To Reproduce:

```
- name: make a new file
  copy: dest=/tmp/foo.txt mode=0644 content="hello world"

- name: check stat of file
  stat: path=/tmp/foo.txt
  register: stat_result

- assert:
    that:
        - "stat_result.stat.mode == 644"
```
##### Expected Results:

Assert task does not fail.
##### Actual Results:

Assert task fails.

**Comment by nilsbunger** [comment]
The mode is correct. 644 is an octal representation, which is 420 in decimal.

If you change the assertion to
    - "{{ '%o' % stat_result.stat.mode }} == 644"

then your assertion passes.

**Comment by jimi-c** [comment]
I think it would actually probably be better to fix the mode to correctly show the octal display in the stat module as follows:

<pre>
diff --git a/library/files/stat b/library/files/stat
index 6e7dcd6..abece4a 100644
--- a/library/files/stat
+++ b/library/files/stat
@@ -88,7 +88,7 @@ def main():
     # back to ansible
     d = {
         'exists'   : True,
-        'mode'     : S_IMODE(mode),
+        'mode'     : "%04o" % S_IMODE(mode),
         'isdir'    : S_ISDIR(mode),
         'ischr'    : S_ISCHR(mode),
         'isblk'    : S_ISBLK(mode),
</pre>

I think users would expect to see the octal representation there.

**Comment by nilsbunger** [comment]
Maybe you're right. But it would be backwards-incompatible to just change it, and there are advantages to the way it is now (easier to do a test like "is this file readable" with bit masking). Maybe a project lead can weigh in on what they want?

**Comment by bcoca** [comment]
add another field returned 'octal_perms'?

**Comment by mpdehaan** [comment]
I think I agree with jimi-c here in that we should fix the glitch.  Everybody thinks about these things in octal.

**Comment by risaacson** [comment]
Closed as a part of #6221.
